### ISSUE TITLE
Proposal: Add blog functionality to innersourcecommons.org

### ISSUE THREAD

**Issue by yuhattor** [InnerSourceCommons/innersourcecommons.org#764]
## Background

Currently, most of the InnerSource Commons working group's artifacts are either hosted in GitHub Issues, Google Docs, or published as formal content such as InnerSource Patterns or the Managing InnerSource book. However, there seems to be a gap for content that doesn't fit these formats or isn't ready for formal publication.

## Proposal

I propose adding a blog functionality to the innersourcecommons.org website, perhaps under innersourcecommons.org/blog. 

This would provide several benefits:

1. A dedicated place to publish content that is valuable but doesn't fit the existing formal formats
2. Ability to share work-in-progress ideas and stories to get feedback
3. Potentially drive more traffic to the InnerSource Commons site as the blog grows
4. Encourage participation from a wider range of contributors
5. Lower the barrier to entry by enabling more casual story sharing, which could feed into patterns, learning paths, Managing InnerSource Project book, etc. 

I have considered several points that we should be aware of.

- The purpose is to allow casual content and unproven ideas to be shared, but this doesn't mean accepting posts of any quality level. 
- Random thoughts and musings are okay occasionally, but should not be the majority of content. Posts should aim to convey concrete experiences where possible.
- Diversity in authorship is important. The same person should not dominate the blog with multiple consecutive posts.
- Respect opinions from those outside the InnerSource community. Encourage related ideas and topics to be posted, even if not 100% aligned with InnerSource. This is a chance to increase connections to various themes, similar to how Agile, Scrum and DevOps are discussed in the XP domain.
- A review process needs to be established to curate content and ensure valuable information is being aggregated, considering the above points.

## Next Steps

1. Discuss the proposal with the InnerSource Commons community and gather feedback
2. If the proposal is accepted, identify the technical requirements and resources needed to enable blog functionality on innersourcecommons.org
3. Create a plan for implementing and launching the blog, including guidelines for content.

**Comment by rrrutledge** [comment]
This idea is great!  I have people ask periodically where is our blog or if we they can write on our blog.

**Comment by rrrutledge** [comment]
Our LinkedIn account can have a blog, too.

**Comment by dellagustin** [comment]
Thank you for creating this issue @yuhattor , I have though about this several times already. It would be great to have a blog in our page.
Adding to your list of requirements, we should be sure to enable users to follow/subscribe to our blog using RSS.

**Comment by yuhattor** [comment]
Thank you!
Certainly RSS is important. I'll have to think about the technical requirements a bit.

I think there are Pros & Cons to posting on LinkedIn. It is very good that the review process is open.
Putting them on .org is more flexible in terms of our brand, platform restrictions. For example, when we want to have multiple contributors .org site allows many customization.

Also from a innersourcecommons.org SEO perspective, it is definitely better to post articles on .org, but from the SEO perspective of the **articles themselves**, LinkedIn may be a better option. Are there any other aspects where LinkedIn is better?

I would also like to find out what kind of content would fit.
👆 I will post this on #general channel

**Comment by clcoffey** [comment]
I think it's a great idea for SEO and spreading new ideas as well as increasing engagement.
One question I would have is, do we care about regular posting and if so, will we have enough contributors and content to maintain a schedule?  

On LinkedIn, I think there are fewer expectations for regular posting while on an official company website, I would expect the posts to follow a schedule. 

A negative when posting a blog/article on the LinkedIn page is that it's harder to attribute credit to the writer unless we link them in the post. 

A few ways to ensure regular posting 
- converting the content covered in the Community calls to blog posts (or having these as a backup)
- creating a list of initial contributors who have ideas to share. 
- creating a backlog of blog posts that can be shared over time.

**Comment by yuhattor** [comment]
Thank you for your comment. You're right.
I think the first 20 or so articles will be fine. We just need to reach out to the members who are currently active.

However, in order to get people to write articles regularly and continuously in the future, we may need to make the following efforts
- Ask the person who wrote the Pattern to write an article including the context of that time.
- By publishing articles about people who have thought about Pattern and their experiences with it that have not yet matured, we may be able to contribute to increasing the number of instances of Pattern.
- Talk to people who were once active in the group and ask them to write articles again.
- Ask pundits who are not related to the inner source, but who have commonality on topics such as Agile, XP, and of course open source, to contribute articles (this may be a challenging one)

We need to make sure we work properly, including the review process.
**If we are going to test it, perhaps we should try a PoC on LinkedIn, which does not require implementation, and consider the degree of operation.**
In the meantime, we have a Blog environment on our .org to migrate the content to.

**Comment by dellagustin** [comment]
> One question I would have is, do we care about regular posting and if so, will we have enough contributors and content to maintain a schedule?

I'm totally ok with not having a schedule, and having the blog as a way put unstructured content out. We have this gap and I would not want to have to rely on a 3rd party proprietary platform (i.e. LinkedIn and the likes) to host our content.
We could mirror content on other platforms, but I think they should be in a place we control.

**Comment by spier** [comment]
I like the idea of having a blog for the ISC.

A recent example where I might have used the blog as a medium to publish on:
The story of the translation of the InnerSource patterns to Galician. We did publish this in our newsletter but nowhere else I believe. It could have been an [official announcement](https://innersourcecommons.org/about/announcements/) as well, however that has a pretty official vibe, and didn't 100% fit.

Having maintained a blog for a company in the past, my questions are:

- Who are eligible authors? e.g. only Members or anybody in slack or anybody we know or ...?
- Who does the reviews? If a blog post is published on innersourcecommons.org it does look kind of official. So we should have a super lean review process, I believe. e.g. could be 4 eye principle between the author and 1 reviewer, where the reviewer has to be an ISC Member or similar.
- We could draw a small decision tree: when to publish on the blog, when in the learning path, when in the patterns, when on ISC announcements, etc. This would help us to identify the distinct purposes of each of these resources. This could happen at any point though, and should not prevent the blog from going live.

I am also in favor of testing the idea somewhere before making any technical investments. At the end, the idea would die due to a lack of content. Therefore it would be good to test how much content we have, that would fit this new medium.

Thanks for pushing this @yuhattor !

**Comment by clcoffey** [comment]
Those are some great insights @spier. It sounds to me that there are a few decisions to be made and actions to be taken before we commit. 
- building a list of writers who would like to contribute content
- starting a list of articles @yuhattor you mentioned 20 which would be great but even 10 would keep the blog going for a while 
- the only reason I would see us testing with a different platform is if we weren't confident we would have a stream of content and we end up losing momentum. If we have a batch of articles to start with, I agree with @dellagustin we shouldn't rely on a third-party platform
- @spier the posting and review process is also important and takes resources. Another question I have here is what would the reviewer look for. Is it just syntax, spelling and making sure no foul language makes its way through or are we looking at relevancy as well?

Should we start a doc building a list of actions and documenting all these decisions? How can we take the next steps to progress this?

**Comment by yuhattor** [comment]
Thanks everyone for the great discussion and insights on this blog proposal!!

As a next step, I propose we do a lightweight proof-of-concept to validate the idea. Based on the discussion, here's what I'm thinking:

1. Identify an initial group of 4-5 InnerSource Commons members and active community participants who are interested in contributing an article. This will help ensure we have a critical mass of content to start.

2. Prioritize inviting our "core" members to nominate topics or people to reach out to for an article. Also consider people who have presented at recent events. 


3. Focus the initial content on a few key themes: 
- Examples and experiences putting patterns into practice 
- Event reports and recaps
- Invited articles from thought leaders in related domains like DevOps
- Technical guides and tool overviews
- Case studies
Deprioritize opinion pieces and open-ended idea generation for now.

4. Implement a lightweight review process, focused on basic quality control (spelling, syntax, no inappropriate content) and general relevance to InnerSource. Full committee review isn't needed at this stage. 

5. Publish the initial batch of articles on the innersourcecommons.org site to validate the technical setup and get preliminary feedback. Ensure RSS is enabled.

6. If the initial articles are well-received, then invest in building out the full blog platform and formal contribution/review processes. 

The key is to start with a focused, curated approach to ensure the initial content is high-quality and on-theme. Over time, we can open it up to a wider range of contributors and topics.

For now, I've done 2 hours hack and made sure that it's relatively easy implementation by using existing fomrats and rss can be implemented.

![Image](https://github.com/InnerSourceCommons/innersourcecommons.org/assets/15963767/bd10bad3-17ec-4e78-b3fb-b2388ceb23ed)
![Image](https://github.com/InnerSourceCommons/innersourcecommons.org/assets/15963767/3f6b33ef-5c93-4634-851e-e73cb5ebf17c)

**Comment by CiaraFlanagan** [comment]
Hey @yuhattor 

I've been following the conversation on slack and have only just remembered that you posted this issue! Thanks for getting the ball rolling.

I am in agreement that there is a bit of a gap in relation to more informal content and I agree that a blog platform could provide that space for thoughts, opinions and learning. 

It's great to see that @clcoffey, @spier and @dellagustin have also shown a lot of interest and energy in this. 

I do want to share some more of my thoughts ... Here they are in no particular order:

1. There's an increasing number of articles on social about InnerSource.  The quality isn't always great ... How does the review process make a decision on what to publish or not ... Particularly if someone specifically requests that we publish their article/blog on the website?

2. My opinion is that if we are to have a blog section/page on the website, that isn't an official InnerSource Commons blog ... I think we need to differentiate it so that it's not necessarily representing the InnerSource Commons Foundation ... What do you all think of something along the lines of InnerSource Community Blogs ... (or something with a subtitle such as articles from our community / InnerSource enthusiasts) ... I don't have this quite right but I'm hoping that you get what I mean ... I think there will be more of a responsibility placed on contributors if this is an official InnerSourceCommons Foundation blog ... 

3. I see the conversation above talked maintainers and reviewers ... The review and requested edit process can be more tricky for written content - especially for content in different languages ... How do we address that? 

4. I'm just checking here ... are people in this conversation interested in making this blog a reality and becoming reviewers/maintainers?  @spier @clcoffey @dellagustin @yuhattor - I'd really like your thoughts on this answer.

5. I think with blog content, we can get very wrapped up in pumping it out on a regular basis.  My viewpoint is, we post quality content when we have it and we leave it when we don't. There's a lot of content out there that we could share ... but just be aware that this wouldn't necessarily be 'from the InnerSource Community'

6. I've really enjoyed reading all your thoughts on this and I'm very keen to learn from your perspectives.  Thanks for getting this moving.

**Comment by yuhattor** [comment]
@CiaraFlanagan I have considered the following.
It is rough, but the point is that we should start small, with about three governance members to begin with, and consider a process of requesting only high quality articles, rather than soliciting articles from anywhere.

## Consideration

> There's an increasing number of articles on social about InnerSource. The quality isn't always great ... How does the review process make a decision on what to publish or not ... Particularly if someone specifically requests that we publish their article/blog on the website?

I think articles should basically be composed by what we request. Not everyone should be able to contribute.
There are limits to our quality control, but I think we need to encourage people to contribute, at the very least, of high quality.

> My opinion is that if there is a blog section/page on the website, then it is not an official InnerSource Commons blog ... I think it is necessary to distinguish that it does not necessarily represent the InnerSource Commons Foundation

I think we need to intentionally indicate that it is a contributed blog. we could have something like blog.innersourcecommons.org, but as an SEO, we should have something like patterns.innersourcecommon.org, blog. innersourcecommons.org, managing.... etc. are all perceived as “completely different sites” by Google's search engine, so it is better not to create a variety of different things on various domains.

So I think there should be a rule to require a preface, for example, “This blog was contributed by Ms. Foo Bar of Microsoft”. Do you think that's enough?

> I see the conversation above talked maintainers and reviewers ... The review and requested edit process can be more tricky for written content - especially for content in different languages ... How do we address that?

I don't think multilingualism is necessary at first.

> I'm just checking here ... are people in this conversation interested in making this blog a reality and becoming reviewers/maintainers? @spier @clcoffey @dellagustin @yuhattor - I'd really like your thoughts on this answer.

I am interested, but this would be a good time to solicit new ones, as also there may be others who are interested.

> My viewpoint is, we post quality content when we have it and we leave it when we don't

I couldn't agree more on this

## Schedule and Rough Plan

```mermaid
gantt
    title InnerSource Commons Blog Implementation Plan
    dateFormat  YYYY-MM-DD
    axisFormat  %Y-%m

    section 3 Month Plan
    Identify initial contributors and topics : a1, 2024-06-01, 30d
    Implement lightweight review process : a2, after a1, 14d
    Publish initial 3-4 articles on innersourcecommons.org : a3, after a2, 30d
    Evaluate initial articles and gather feedback : a4, after a3, 14d
    
    section 6 Month Plan
    Build out full blog platform on .org if initial articles succeed : b1, after a4, 60d
    Formalize contribution and review processes : b2, after b1, 30d
    Expand contributor base and content types : b3, after b2, 60d
    Implement content calendar with 1 post/month : b4, after b3, 30d

    section 1 Year Plan
    Grow blog to 2-4 posts/month : c1, after b4, 180d
    Establish blog as go-to resource for InnerSource content : c2, after c1, 180d
    Continuously improve based on metrics and feedback : c3, after c1, 180d
```

Key points:

- Start with a focused, curated approach with 4-5 initial contributors to ensure high-quality, on-theme content
- Implement a lightweight review process for basic quality control and relevance
- Publish initial articles on innersourcecommons.org to validate concept and get feedback 
- If initial articles succeed, invest in building out full blog platform and formalizing processes
- Prioritize quality over quantity and regular cadence, especially early on
- Expand contributor base, content types, and posting frequency over time as blog matures
- Continuously gather feedback and metrics to guide improvement

Measuring success:
- Engagement metrics: page views, time on page, comments, social shares
- Contributor growth and diversity 
- Qualitative feedback from community
- Inbound links and references to blog content from other sites
- Impact on InnerSource adoption (survey data, anecdotal reports)

Governance:
- Nominate 2-3 initial blog maintainers/editors to oversee content and processes
- Maintainers have authority to make content decisions in alignment with blog's mission
- Revisit governance approach after 6-12 months and expand team if needed


By starting small, measuring results, and iterating, the InnerSource Commons blog can grow into a valuable resource for the community while managing risk and effort. The focus should be on surfacing high-quality, insightful content that complements existing resources like patterns and case studies.

**Comment by rrrutledge** [comment]
Yes to the above!

And …

Start this as a working group with an initial set of Trusted Committers, who should all be ISC members.

And …

https://github.com/InnerSourceCommons/foundation-governance/blob/master/how-to/create-working-group.md

**Comment by yuhattor** [comment]
Will talk with @CiaraFlanagan to talk about how to align to marketing wg on June 12th. Please ping us if you want to join us:)
Anyways, It's just a little casual 30m talk, then I'm sure I'll get you all more involved 😉

**Comment by CiaraFlanagan** [comment]
Just had a great call with @yuhattor about experimenting with an InnerSource blog.

### Vision
We discussed a vision for what it might look like.  Yuki had something in mind along the lines of the [GitHub](https://github.blog/2023-11-08-the-state-of-open-source-and-ai/) and [Spotify for Backstage](https://backstage.spotify.com/discover/blog/?page=7#all) blogs.

### Audience
We talked briefly about an audience for our blogs. Yuki mentioned wanting to use the blog to empower people working in scrum and agile to learn more about InnerSource practices. (Is that correct, Yuki? Please correct me if I'm wrong.)

We also talked about using the blog to educate people to attract more people to the community and learn more about InnerSource.

### Content
We discussed a few approaches:
- linking in with members of the community who we know have already posted blogs on LinkedIn to use their existing content and discuss adding more.
- Using ChatGPT to write blogs about InnerSource artefacts, books and patterns
- Inviting those who have written patterns to write a short blog about them
- Keep some entries simple - e.g. the concept of a very short blog article that offers InnerSource 'hacks'

### Frequency of new content
Yuki shared learning about the experience of blog creation in the Japanese community. This will be discussed again.

### The 'how'
We talked about what's required in terms of maintaining and reviewing.

Ciara volunteered to review content but we agreed that we need more people to get involved.

### Next steps
Ciara will set up a new channel for the InnerSource blog and will get in touch with others who may be interested.

**Comment by CiaraFlanagan** [comment]
@yuhattor 

Please add anything that I may have left out. And please feel free to correct anything.

**Comment by CiaraFlanagan** [comment]
> Yes to the above!
> 
> And …
> 
> Start this as a working group with an initial set of Trusted Committers, who should all be ISC members.
> 
> And …
> 
> https://github.com/InnerSourceCommons/foundation-governance/blob/master/how-to/create-working-group.md

Hi @rrrutledge 

We're thinking of aligning this work with the work of the existing Marketing Working Groups and capturing any notes from meetings here on GitHub. This will make the process more lightweight.

**Comment by rrrutledge** [comment]
This is fine if the initial set of Trusted Committers can make the marketing working group meeting calls.

**Comment by yuhattor** [comment]
@CiaraFlanagan

One quick thought. 
We would write a outcome based blog where we first create a set of 30 or so categories we want to write about, or a set of hypothetical titles, and then invite or nominate people to write about them!
That way, we can decide "who will review it" first.
When a hypothetical theme is decided upon, we can then vote on it,
That way, at least the timeline ahead is clear.

Then a document could be created for the purpose of shipping here in advance, including ISPO working groups, pattern ideas, etc.
Once the purpose is clear, it will be easier for everyone to participate.

It is too hard for the Trusted Committer to take on a fixed role of reviewer.
Wouldn't it be better if the Trustee Committer's role was not to review, but to be responsible for scheduling, managing the quality of overall content, developing direction, and assigning reviewers


---
By the way, here's the sample implementation
https://github.com/InnerSourceCommons/innersourcecommons.org/pull/775

**Comment by CiaraFlanagan** [comment]
### **Meeting Notes from June 20th**

### Attendees
@yuhattor 
@CiaraFlanagan 
Apostolos Kritikos
@rrrutledge 
Michael Basil

### Objective of Meeting
This is a casual meeting and brainstorming session about a potential blog.

@yuhattor has also talked to @jeffabailey about this.

### Agenda

- We need to consider differences between the blog and other offerings e.g. ISPO Working groups etc
- Increasing number of authors for content

### InnerSource blog and the ISPO Working Group

- Members of the group discussed how they discovered / got into InnerSource and also the ISC Foundation.
- The discussion also touched on the onboarding process in ISC, 'orientation supply', and the need to support new people to become involved and how to facilitate synchronous exchange and community collaboration.
- The blog can be the 'storytelling' aspect that breathes life into InnerSource values and shares a very practical approach to sharing community learning and community.

### What role can the InnerSource blog play?
- Can this serve as a gateway to the community? 
- It can also attract new members with different DevOps / Agile use cases of how InnerSource was applied
- InnerSource blogs can modelled and seen as mentorship by key figures in the community
- Helps create a voice for the ISC community
- 'Our place to tell stories'  + storytelling as a way of teaching (Sometimes we lead with the story and then the lessons learned are outlined by the storyteller or drawn out by those listening)
- The blog is a tool we can use to fulfil the mission of the Foundation ... to reach people through a written medium.
- Summaries of what is happening in the sector ... Allow for quick posts / short form that are easily consumable by others ...?
- Our patterns and using the blog to tie into existing patterns is something core to our community.

### Defining the blog

- Group talked about the difference between our InnerSource stories page and the blog
- Ciara also noted the importance of clear differentiation between the Stories page, the newsletter and the Blog
- Is it the Foundation blog or another guest speaking about InnerSource? 
- **In summary: Blogs should tie into our patterns and other resources, they must be approved by the Foundation and align with the vision and mission of the Foundation. The posts can be written by members of the community (subject to approval) or from the Foundation themselves.**


### Quality and Maintenance
- We need a plan and Trusted Committers
- How do we maintain it?
- Suggest that we maintain this in a code repo ...via GitHub actions?

**Comment by yuhattor** [comment]
The idea was to use this for organizing practices and information, but I've decided to put this on hold for now. On the other hand, considering that InnerSourceCommons.org currently functions mainly as the Foundation's site, rather than writing casual content, it might be better to redefine it as a place for "Foundation records" and "facts."

I realized something serious.
For articles, the Foundation's 10th anniversary article, or the InnerSource 25th anniversary article - "currently, there's no place to write them."
To address this, I want to reorganize the blog for now.

What should be in the ISC Blog [fact-based content]:
- InnerSource 25th anniversary articles written by members
- ISC 10th anniversary articles written by members
- Gathering event reports
- Conference presentation reports by regular members
- Carefully selected fact-based company case studies
- Announcements of new patterns (those that have reached maturity or are growing and seeking participation)
- ISPO docs reports (reports on achievements)
- Articles by people invited by members (those providing endorsements)

What should not be included:
- Brief content that should go to the newsletter
- Unproven practices like "this is what happened when we implemented it"
- Hypotheses about InnerSource
- Hero of the month

Things that may happen in the future but won't happen initially (careful discussion needed):
- Participatory articles by non-Foundation members
- Republishing of newsletters

**Comment by yuhattor** [comment]
I will once close this issue and start the new thread again
